### PR TITLE
Fix#99

### LINKED ISSUE #99: Bulk Data Insertion in Fusion Auth
The `addbottoregistry` and `adddevicetoregistry` APIs in Fusion Auth currently process data individually, resulting in slow processing times for large data sets exceeding 80,000 entries. To improve efficiency and accommodate bulk data queries, it is necessary to modify these APIs to support batch processing.

### PR DESCRIPTION
**Pull request by RyanWalker277**
Closes #99

### CODE REVIEW

File: v2/uci/src/modules/user-segment/fusionauth/fusionauth.service.spec.ts [hunk 1]
```diff
@@ -4,19 +4,11 @@ import FusionAuthClient, { ApplicationRequest, RegistrationRequest, UUID } from
...
 class MockFusionAuthClient {
   callCounter = 0;
 
   async createApplication(_botID: string, _request: ApplicationRequest): Promise<any> {
-    if (this.callCounter == 0) {
-      this.callCounter = 1;
-      return Promise.reject('Test reject');
-    }
-    else {
-      this.callCounter = 0;
```
Comment (chinmoy12c) [inline v2/uci/src/modules/user-segment/fusionauth/fusionauth.service.spec.ts:17]: Please revert this change, this logic was put in place for `FusionAuthClient` retries, or find a workaround logic for this.

**Comment by RyanWalker277** [comment]
Completed the requested changes, @chinmoy12c kindly review.

File: src/modules/user-segment/fusionauth/fusionauth.service.ts [hunk 2]
```diff
@@ -82,6 +82,24 @@ export class DeviceManagerService {
...
+        }
+      });
+
+      await Promise.all(promises);
+      console.log('All bots added to the registry.');
+    } catch (error) {
+      console.log('Error occurred while adding bots:', error);
+      throw new InternalServerErrorException('Failed to add all bots to the registry.');
+    }
+  };
```
[the remark this answers is not in the document]
Comment (RyanWalker277) [reply src/modules/user-segment/fusionauth/fusionauth.service.ts:101]: Sure! Will update tue function accordingly.

**Comment by RyanWalker277** [comment]
Hey @chinmoy12c, completed the necessary changes. KIndly review
